### PR TITLE
Show Application Principal ID during auth

### PR DESCRIPTION
**Pull request by nmattia**
## Description

This re-adds the functionality introduced by
https://github.com/dfinity/internet-identity/pull/325 but makes the
Principal ID less visible.

**NOTE**: I've added a tooltip. This is what I would have needed to understand the process the first time I authenticated with II as a user.

https://user-images.githubusercontent.com/6930756/128024280-79391604-f25d-460f-8c60-146de41fc8c3.mov

This works by calling `get_principal()` during the auth process and
thread the resulting principal ID to the authentication confirmation
screen.

## Motivation and Context

Currently it seems difficult for users to understand that every front end application they use on the IC with II will perceive them under a different principal. The first attempt for mitigating this is presenting the actual principal that will be used upon redirecting the user.

## How Has This Been Tested?

Manual front-end test with a local deployment.

## Screenshots (if appropriate):

Screenshot 2021-08-03 at 15 33 52
Screenshot 2021-08-03 at 15 34 19

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/frontend/src/auth.ts [hunk 1]
```diff
@@ -88,7 +88,9 @@ async function handleAuthRequest(
   request: AuthRequest,
   hostname: FrontendHostname
 ): Promise<AuthResponse> {
-  if (!(await confirmRedirect(hostname))) {
+  const userPrincipal = await connection.getPrincipal(userNumber, hostname);
```
Comment (kritzcreek) [inline src/frontend/src/auth.ts:91]: Do we need to cover this with a spinner? Query calls are fast, but depending on your connection you might still end up with an "unresponsive" app.

**Review by kritzcreek — CHANGES_REQUESTED** [review]
Thanks, changes look great in general, and I love the tooltip.

Just have a couple of nits.

File: src/frontend/src/auth.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/auth.ts:91]: Are you thinking the "astronaut" kind of spinner or a new kind?

Generally this makes sense though!

Comment (kritzcreek) [reply src/frontend/src/auth.ts:91]: Yeah the astronaut. I do the same when loading the data for the `manage` page with a query

File: src/frontend/src/flows/confirmRedirect.ts [hunk 2]
```diff
@@ -1,15 +1,51 @@
 import { html, render } from "lit-html";
 import { FrontendHostname } from "../../generated/internet_identity_types";
 
-const pageContent = (hostName: string) => html`
+// We show a tooltip giving more information about the "application-specific"
+// nature of pseudonyms. This is the text that gets displayed when hovering.
+const textTooltip = `Internet Identity issues a unique "pseudonym" for each application requesting
+authentication. Since this pseudonym is unique for each application you
+authenticate for, it cannot be used to track you.
```
Comment (nmattia) [inline src/frontend/src/flows/confirmRedirect.ts:8]:
```suggestion
authenticate for, applications cannot use it to track you.
```

Suggestion from @Dfinity-Bjoern

File: src/frontend/src/auth.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/auth.ts]: Done! PTAL.

**Review by kritzcreek — APPROVED** [review]

**Pull request merged**